morango/utils/uuids.py

```diff
@@ -104,21 +106,39 @@ def save(self, *args, **kwargs):
         super(UUIDModelMixin, self).save(*args, **kwargs)
 
 
+class DatabaseManager(models.Manager):
+
+    def create(self, **kwargs):
+
+        # set current flag to false for all database_id models
+        DatabaseIDModel.objects.update(current=False)
+        super(DatabaseManager, self).create(**kwargs)
```

Comment:
Safest to do these two operations within some sort of transaction, so we never have a limbo state with no "current" database ID.